File: GitToolBox/src/main/java/zielu/gittoolbox/ui/actions/InlineBlameToggleAction.java
package zielu.gittoolbox.ui.actions;

import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.actionSystem.ToggleAction;
import org.jetbrains.annotations.NotNull;
import zielu.gittoolbox.config.AppConfig;
import zielu.gittoolbox.config.ConfigUtil;
import zielu.gittoolbox.config.GitToolBoxConfig2;

public class InlineBlameToggleAction extends ToggleAction {

  @Override
  public boolean isSelected(@NotNull AnActionEvent e) {
    return AppConfig.get().getShowEditorInlineBlame();
  }

  @Override
  public void setSelected(@NotNull AnActionEvent e, boolean state) {
    GitToolBoxConfig2 current = AppConfig.get();
    if (current.getShowEditorInlineBlame() != state) {
      ConfigUtil.saveAppSettings(config -> config.setShowEditorInlineBlame(state));
    }
  }
}

File: GitToolBox/src/main/kotlin/zielu/gittoolbox/ui/actions/InlineBlameToggleAction.kt
package zielu.gittoolbox.ui.actions

import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.actionSystem.ToggleAction
import zielu.gittoolbox.config.AppConfig
import zielu.gittoolbox.util.AppUtil.modifySettingsSaveAndNotify

internal class InlineBlameToggleAction : ToggleAction() {
  override fun isSelected(e: AnActionEvent): Boolean = AppConfig.get().showEditorInlineBlame

  override fun setSelected(e: AnActionEvent, state: Boolean) {
    val current = AppConfig.get()
    if (current.showEditorInlineBlame != state) {
      modifySettingsSaveAndNotify { it.showEditorInlineBlame = state }
    }
  }
}
